decryptor/mysql/response_proxy.go

```diff
@@ -397,49 +459,95 @@ func (handler *Handler) ProxyClientConnection(ctx context.Context, errCh chan<-
 	}
 }
 
-func (handler *Handler) handleStatementExecute(ctx context.Context, packet *Packet) error {
-	stmtID := binary.LittleEndian.Uint32(packet.GetData()[1:])
+func getParamsCount(stmt sqlparser.Statement) (int, error) {
+	var paramsCount int
+	err := sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+		switch nodeType := node.(type) {
+		case *sqlparser.SQLVal:
+			if bytes.HasPrefix(nodeType.Val, []byte(":v")) {
+				paramsCount++
+			}
+		}
+		return true, nil
+	}, stmt)
+	return paramsCount, err
+}
 
-	log := handler.logger.WithField("proxy", "client").WithField("statement", stmtID)
-	log.Debug("Statement Execute")
+func (handler *Handler) handleStatementExecute(ctx context.Context, packet *Packet) (uint32, error) {
+	var (
+		paramsNumber int
+		statement    sqlparser.Statement
+		stmtID       = binary.LittleEndian.Uint32(packet.GetData()[1:])
```

Comment:
let's check that GetData() returns at least 2 bytes before the call?) to avoid access out of bounds